File: 0013-roman-to-integer/0013-roman-to-integer.java
import java.util.*;

class Solution {
    public int romanToInt(String s) {
        int ret = 0;
        for(int i = 0; i < s.length(); ++i) {
            char ch = s.charAt(i);
            if(ch == 'I') {
                if(s.indexOf("IV", i) == i) {
                    ret += 4;
                    ++i;
                } else if(s.indexOf("IX") == i) {
                    ret += 9;
                    ++i;
                } else ret += 1;
            } else if(ch == 'V') {
                ret += 5;
            } else if(ch == 'X') {
                if(s.indexOf("XL", i) == i) {
                    ret += 40;
                    ++i;
                } else if(s.indexOf("XC", i) == i) {
                    ret += 90;
                    ++i;
                } else ret += 10;
            } else if(ch == 'L') {
                ret += 50;
            } else if(ch == 'C') {
                if(s.indexOf("CD", i) == i) {
                    ret += 400;
                    ++i;
                } else if(s.indexOf("CM", i) == i) {
                    ret += 900;
                    ++i;
                } else ret += 100;
            } else if(ch == 'D') {
                ret += 500;
            } else if(ch == 'M') {
                ret += 1000;
            }
        }
        
        return ret;
    }
}

File: 0013-roman-to-integer/0013-roman-to-integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        roman = {'I': 1,'V': 5,'X': 10,'L': 50,'C': 100,'D': 500,'M': 1000}
        length = len(s)

        ans, i = 0, 0
        while i < length:
            if i + 1 < length and roman[s[i]] < roman[s[i + 1]]:
                ans += roman[s[i + 1]] - roman[s[i]]
                i += 2
            else:
                ans += roman[s[i]]
                i += 1
        return ans
